src/projects/repeat_resolution/mdbg.hpp

```diff
@@ -0,0 +1,200 @@
+//
+// Created by Andrey Bzikadze on 11/10/21.
+//
+
+#pragma once
+
+#include "error_correction/multiplicity_estimation.hpp"
+#include "mdbg_topology.hpp"
+#include "paths.hpp"
+
+namespace repeat_resolution {
+
+class MultiplexDBG
+    : public graph_lite::Graph<
+        /*typename NodeType=*/RRVertexType,
+        /*typename NodePropType=*/RRVertexProperty,
+        /*typename EdgePropType=*/RREdgeProperty,
+        /*EdgeDirection direction=*/graph_lite::EdgeDirection::DIRECTED,
+        /*MultiEdge multi_edge=*/graph_lite::MultiEdge::ALLOWED,
+        /*SelfLoop self_loop=*/graph_lite::SelfLoop::ALLOWED,
+        /*Map adj_list_spec=*/graph_lite::Map::UNORDERED_MAP,
+        /*Container neighbors_container_spec=*/
+                              graph_lite::Container::MULTISET> {
+    friend class MultiplexDBGIncreaser;
+    RRPaths *rr_paths;
+    uint64_t next_edge_index{0};
+    uint64_t next_vert_index{0};
+    uint64_t n_iter{0};
+    uint64_t start_k{1};
+    bool contains_rc = true;
+
+    static std::vector<SuccinctEdgeInfo>
```

Comment:
That would be nice to have SuccinctEdgeInfo and all related stuff to be completely hidden from hpp files but I guess you need this to for tests, right? Otherwise this static method could be just function in cpp, SuccintEdgeInfo also could be moved to cpp completely and corresponding constructors for MultiplexDBG removed.

Reply:
Yes, I unfortunately need this for tests because it is difficult to mock class `SparseDBG`. I think mocking of all main data structures deserves a separate discussion. Please follow up if you want a specific action on my side.

Reply:
I added an [issue](https://github.com/AntonBankevich/LJA/issues/10), so we don't forget about this